graylog2-web-interface/src/components/streamrules/DetailsStreamRule.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import HumanReadableStreamRule from 'components/streamrules/HumanReadableStreamRule';
+import { useStore } from 'stores/connect';
+import { Icon } from 'components/common';
+import { Button } from 'components/bootstrap';
+import { isPermitted } from 'util/PermissionsMixin';
+import StreamRuleModal from 'components/streamrules/StreamRuleModal';
+import UserNotification from 'util/UserNotification';
+import { StreamRulesInputsActions, StreamRulesInputsStore } from 'stores/inputs/StreamRulesInputsStore';
+import { StreamRulesStore } from 'stores/streams/StreamRulesStore';
+import type { StreamRule as StreamRuleTypeDefinition, Stream, StreamRule } from 'stores/streams/StreamsStore';
+
+import useCurrentUser from '../../hooks/useCurrentUser';
+
+const ActionButtonsWrap = styled.span`
+  margin-right: 6px;
+  float: right;
+`;
+
+type Props = {
+  matchData?: {
+    matches: boolean,
+    rules: { [id: string]: false },
+  },
+  stream: Stream,
+  onDelete: (ruleId: string) => void,
+  onSubmit: (ruleId: string, data: unknown) => void,
+  streamRule: StreamRuleTypeDefinition
+}
+
+const DetailsStreamRule = ({ stream, streamRule, onSubmit, onDelete }: Props) => {
+  const { permissions } = useCurrentUser();
+  const [showStreamRuleForm, setShowStreamRuleForm] = useState(false);
+  const { inputs } = useStore(StreamRulesInputsStore);
+
+  useEffect(() => {
+    StreamRulesInputsActions.list();
+  }, []);
+
+  const _onEdit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    setShowStreamRuleForm(true);
+  };
+
+  const _onDelete = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    /* TODO: Replace with custom confirmation dialog */
+    // eslint-disable-next-line no-alert
+    if (window.confirm('Do you really want to delete this stream rule?')) {
+      StreamRulesStore.remove(stream.id, streamRule.id, () => {
```

Comment:
As you wrote it looks better if we use our ConfirmDialog instead of browser alerts